Author: Aaron M. Ucko
Last-Update: 2023-01-15 20:56:15
Description: Label tests "online" to have an easy option to run/ignore them properly

Index: b/ngs/ngs-sdk/examples/CMakeLists.txt
===================================================================
--- a/ngs/ngs-sdk/examples/CMakeLists.txt
+++ b/ngs/ngs-sdk/examples/CMakeLists.txt
@@ -52,7 +52,7 @@ if ( NOT TOOLS_ONLY )
         add_test( NAME SlowTest_NGS_Run_and_diff COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/run_and_diff.sh ${TESTBINDIR}
                 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
         set_tests_properties( SlowTest_NGS_Run_and_diff
-            PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R"
+            PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS "online"
         )
     endif()
 
Index: b/ngs/ngs-java/examples/CMakeLists.txt
===================================================================
--- a/ngs/ngs-java/examples/CMakeLists.txt
+++ b/ngs/ngs-java/examples/CMakeLists.txt
@@ -56,6 +56,8 @@ if ( Java_FOUND )
                     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
                     COMMAND_EXPAND_LISTS
                 )
+                set_tests_properties(Test_NgsJavaExamples_${TestName}
+                    PROPERTIES LABELS "online")
             endmacro()
 
             RunAndDiff( FragTest "ERR225922 10000 2" )
Index: b/ngs/ngs-python/examples/CMakeLists.txt
===================================================================
--- a/ngs/ngs-python/examples/CMakeLists.txt
+++ b/ngs/ngs-python/examples/CMakeLists.txt
@@ -44,6 +44,7 @@ if( Python3_EXECUTABLE )
                 set_tests_properties( Test_NgsPythonExamples_${TestName}
 
                     PROPERTIES ENVIRONMENT "NGS_PY_DOWNLOAD_LIBRARY=0;PYTHONUSERBASE=${PythonUserBase};NGS_PY_LIBRARY_PATH=${LIBDIR}" DEPENDS Test_NGS_Python_install
+                    LABELS "online"
                     )
             endmacro()
 
Index: b/test/internal/align/CMakeLists.txt
===================================================================
--- a/test/internal/align/CMakeLists.txt
+++ b/test/internal/align/CMakeLists.txt
@@ -31,3 +31,4 @@ endif()
 include_directories( ${VDB_INTERFACES_DIR}/ext/ ) # zlib.h
 include_directories( ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/../ncbi-vdb/interfaces/ext/ )
 AddExecutableTest( SlowTest_Align_estimator "pileup_estimator" "ncbi-ngs-c++;ngs-c++;${COMMON_LINK_LIBRARIES};${COMMON_LIBS_READ}" "" )
+set_tests_properties(SlowTest_Align_estimator PROPERTIES LABELS "online")
Index: b/test/loaders/kar/CMakeLists.txt
===================================================================
--- a/test/loaders/kar/CMakeLists.txt
+++ b/test/loaders/kar/CMakeLists.txt
@@ -30,6 +30,7 @@ if ( EXISTS "${DIRTOTEST}/kar${EXE}" )
             ${CMAKE_COMMAND} -E env ${CONFIGTOUSE}=/ NCBI_VDB_QUALITY=R
             bash -c "./kar-ntest.sh ${DIRTOTEST}/kar ${DIRTOTEST}/prefetch"
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Kar PROPERTIES LABELS "online")
 
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Kar-asan
Index: b/test/external/kdbmeta/CMakeLists.txt
===================================================================
--- a/test/external/kdbmeta/CMakeLists.txt
+++ b/test/external/kdbmeta/CMakeLists.txt
@@ -28,6 +28,7 @@ if ( EXISTS "${DIRTOTEST}/kdbmeta${EXE}"
     add_test( NAME Test_Kdbmeta
           COMMAND bash -c "${DIRTOTEST}/kdbmeta SRR053325 > /dev/null"
           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Kdbmeta PROPERTIES LABELS "online")
 
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Kdbmeta-asan
Index: b/test/external/ngs/ngs-java/CMakeLists.txt
===================================================================
--- a/test/external/ngs/ngs-java/CMakeLists.txt
+++ b/test/external/ngs/ngs-java/CMakeLists.txt
@@ -81,5 +81,6 @@ if ( Java_FOUND )
 	    ${Java_JAVA_EXECUTABLE} -cp "${CLASSPATH}" "-Djava.library.path=${LIBPATH}" ${VARS} org.junit.runner.JUnitCore ngs_test_CSRA1
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
     )
+    set_tests_properties(SlowTest_NgsJava_CSRA1 PROPERTIES LABELS "online")
 
 endif()
Index: b/test/external/ngs-pileup/CMakeLists.txt
===================================================================
--- a/test/external/ngs-pileup/CMakeLists.txt
+++ b/test/external/ngs-pileup/CMakeLists.txt
@@ -31,6 +31,7 @@ set( NGS_PILEUP_LIBS
     ${COMMON_LIBS_READ}
 )
 AddExecutableTest( Test_Ngs_Pileup "test-ngs-pileup.cpp" "${NGS_PILEUP_LIBS}" "" )
+set_tests_properties(Test_Ngs_Pileup PROPERTIES LABELS "online")
 
 if( NOT WIN32 )
 if ( EXISTS "${DIRTOTEST}/ngs-pileup${EXE}" )
@@ -43,6 +44,7 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     add_test( NAME Test_Ngs_Pileup_2
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 2.0 SRR1166393
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Ngs_Pileup_2 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Ngs_Pileup_2-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 2.0 SRR1166393
@@ -56,6 +58,7 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     add_test( NAME Test_Ngs_Pileup_3
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 3.0 ERR334733
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Ngs_Pileup_3 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Ngs_Pileup_3-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 3.0 ERR334733
@@ -68,6 +71,7 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     add_test( NAME Test_Ngs_Pileup_4
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 4.0 SRR1068006
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Ngs_Pileup_4 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Ngs_Pileup_4-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 4.0 SRR1068006
@@ -81,6 +85,7 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     add_test( NAME SlowTest_Ngs_Pileup_5
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 5.0 SRR619510
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(SlowTest_Ngs_Pileup_5 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME SlowTest_Ngs_Pileup_5-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 5.0 SRR619510
@@ -95,11 +100,13 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     # add_test( NAME Test_Ngs_Pileup_6
     #           COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 6.0 SRR1251988
     #           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    # set_tests_properties(Test_Ngs_Pileup_6 PROPERTIES LABELS "online")
 
     # ION_TORRENT
     add_test( NAME Test_Ngs_Pileup_7
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 7.0 SRR1186012
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Ngs_Pileup_7 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Ngs_Pileup_7-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 7.0 SRR1186012
@@ -114,17 +121,20 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
     # add_test( NAME Test_Ngs_Pileup_8.1
     #           COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 8.1 SRR556739 -r chrY
     #           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    # set_tests_properties(Test_Ngs_Pileup_8.1 PROPERTIES LABELS "online")
 
     #TODO: fix
     # COMPLETE_GENOMICS, circular reference
     # add_test( NAME Test_Ngs_Pileup_8.2
     #           COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 8.2 SRR556739 -r chrM
     #           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    # set_tests_properties(Test_Ngs_Pileup_8.2 PROPERTIES LABELS "online")
 
     #:19900-20022
     add_test( NAME Test_Ngs_Pileup_9
             COMMAND ./runtestcase.sh ${BINDIR} "" ${CMAKE_CURRENT_SOURCE_DIR} 9.0 SRR341578 -r NC_011752.1
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Ngs_Pileup_9 PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Ngs_Pileup_9-asan
                 COMMAND ./runtestcase.sh ${BINDIR} "-asan" ${CMAKE_CURRENT_SOURCE_DIR} 9.0 SRR341578 -r NC_011752.1
@@ -136,4 +146,4 @@ if ( EXISTS "${DIRTOTEST}/ngs-pileup${EX
 else()
     message(WARNING "${DIRTOTEST}/ngs-pileup${EXE} is not found. The corresponding tests are skipped." )
 endif()
-endif()
\ No newline at end of file
+endif()
Index: b/test/external/ngs/ngs-python/CMakeLists.txt
===================================================================
--- a/test/external/ngs/ngs-python/CMakeLists.txt
+++ b/test/external/ngs/ngs-python/CMakeLists.txt
@@ -33,6 +33,7 @@ if( Python3_EXECUTABLE )
             PROPERTIES
                 DEPENDS Test_NGS_Python_install
                 ENVIRONMENT "LD_LIBRARY_PATH=${NCBI_VDB_LIBDIR};NCBI_VDB_QUALITY=R;NGS_PY_LIBRARY_PATH=${LIBDIR};PYTHONUSERBASE=${PythonUserBase}"
+                LABELS "online"
         )
     elseif( WIN32 )
         add_test( NAME SlowTest_NgsPython
Index: b/test/external/prefetch/CMakeLists.txt
===================================================================
--- a/test/external/prefetch/CMakeLists.txt
+++ b/test/external/prefetch/CMakeLists.txt
@@ -38,6 +38,8 @@ if ( EXISTS "${DIRTOTEST}/prefetch${EXE}
 	add_test( NAME Test_Prefetch_urls_and_accs
 		COMMAND perl urls_and_accs.pl ${DIRTOTEST} ${BINDIR} prefetch
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+        set_tests_properties(Test_Prefetch_urls_and_accs
+            PROPERTIES LABELS "online")
 	if( RUN_SANITIZER_TESTS )
 		add_test( NAME Test_Prefetch_urls_and_accs-asan
 			COMMAND perl urls_and_accs.pl ${DIRTOTEST} ${BINDIR} prefetch-asan
@@ -50,6 +52,8 @@ if ( EXISTS "${DIRTOTEST}/prefetch${EXE}
 	add_test( NAME Test_Prefetch_out_dir_and_file
 		COMMAND perl out_dir_and_file.pl ${DIRTOTEST} ${BINDIR} prefetch
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+        set_tests_properties(Test_Prefetch_out_dir_and_file
+            PROPERTIES LABELS "online")
 	if( RUN_SANITIZER_TESTS )
 		add_test( NAME Test_Prefetch_out_dir_and_file-asan
 			COMMAND perl out_dir_and_file.pl ${DIRTOTEST} ${BINDIR} prefetch-asan
@@ -64,6 +68,7 @@ if ( EXISTS "${DIRTOTEST}/prefetch${EXE}
             ${CMAKE_COMMAND} -E env ${CONFIGTOUSE}=/
             bash -c "sh runtests.sh ${CMAKE_BINARY_DIR}/test-bin/SlowTest_Prefetch ${DIRTOTEST} prefetch"
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+        set_tests_properties(SlowTest_Prefetch_dflt PROPERTIES LABELS "online")
 	if( RUN_SANITIZER_TESTS )
 		add_test( NAME SlowTest_Prefetch_dflt-asan
 			COMMAND
@@ -80,6 +85,9 @@ if ( EXISTS "${DIRTOTEST}/prefetch${EXE}
 	add_test( NAME SlowTest_Prefetch_1GB
 		COMMAND perl ncbi1GB.pl ${BINDIR} prefetch
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+        # Trivially succeeds in the absence of Aspera ascp, but any
+        # meaningful testing would require online resources regardless.
+        set_tests_properties(SlowTest_Prefetch_1GB PROPERTIES LABELS "online")
 	if( RUN_SANITIZER_TESTS )
 		add_test( NAME SlowTest_Prefetch_1GB-asan
 			COMMAND perl ncbi1GB.pl ${BINDIR} prefetch-asan
@@ -92,6 +100,8 @@ if ( EXISTS "${DIRTOTEST}/prefetch${EXE}
 	add_test( NAME SlowTest_Prefetch_vdbcache
 		COMMAND perl vdbcache.pl ${BINDIR} prefetch
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+        set_tests_properties(SlowTest_Prefetch_vdbcache PROPERTIES
+                             LABELS "online")
 	if( RUN_SANITIZER_TESTS )
 		add_test( NAME SlowTest_Prefetch_vdbcache-asan
 			COMMAND perl vdbcache.pl ${BINDIR} prefetch-asan
Index: b/test/external/sra-pileup/CMakeLists.txt
===================================================================
--- a/test/external/sra-pileup/CMakeLists.txt
+++ b/test/external/sra-pileup/CMakeLists.txt
@@ -43,6 +43,7 @@ if ( EXISTS "${DIRTOTEST}/sra-pileup${EX
                 bash -c "./runtests-slow.sh ${DIRTOTEST} ${Python3_EXECUTABLE}"
             WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         )
+        set_tests_properties(SlowTest_SraPileup PROPERTIES LABELS "online")
     endif()
 
 else()
Index: b/test/external/sra-stat/CMakeLists.txt
===================================================================
--- a/test/external/sra-stat/CMakeLists.txt
+++ b/test/external/sra-stat/CMakeLists.txt
@@ -55,6 +55,8 @@ if ( EXISTS "${DIRTOTEST}/sra-stat${EXE}
 			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
 	endif()
 
+        set_tests_properties(Test_Sra_stat SlowTest_Sra_stat
+            PROPERTIES LABELS "online")
 else()
 	message(WARNING "${DIRTOTEST}/sra-stat${EXE} is not found. The corresponding tests are skipped." )
 endif()
Index: b/test/external/srapath/CMakeLists.txt
===================================================================
--- a/test/external/srapath/CMakeLists.txt
+++ b/test/external/srapath/CMakeLists.txt
@@ -42,6 +42,7 @@ if ( NOT WIN32 )
             ${CMAKE_COMMAND} -E env VDB_CONFIG=../..
             bash -c "./test-srapath.sh ${DIRTOTEST} srapath"
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
+    set_tests_properties(Test_Srapath PROPERTIES LABELS "online")
     if( RUN_SANITIZER_TESTS )
         add_test( NAME Test_Srapath-asan
             COMMAND
Index: b/test/internal/vdb-diff/CMakeLists.txt
===================================================================
--- a/test/internal/vdb-diff/CMakeLists.txt
+++ b/test/internal/vdb-diff/CMakeLists.txt
@@ -65,6 +65,10 @@ if( NOT WIN32 )
 			COMMAND test_failure.sh "${DIRTOTEST}" ${ACCESSION} vdb-diff-tsan
 			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
 	endif()
+
+        set_tests_properties(
+            Test_VDB_Diff_Check_success Test_VDB_Diff_Check_failure
+            PROPERTIES LABELS "online")
 endif()
 
 else()
Index: b/test/external/vdb-dump/CMakeLists.txt
===================================================================
--- a/test/external/vdb-dump/CMakeLists.txt
+++ b/test/external/vdb-dump/CMakeLists.txt
@@ -66,4 +66,5 @@ if ( NOT WIN32 )
         )
     endif()
 
+    set_tests_properties(Test_Vdb_dump PROPERTIES LABELS "online")
 endif()
Index: b/libs/ngs-vdb/test/CMakeLists.txt
===================================================================
--- a/libs/ngs-vdb/test/CMakeLists.txt
+++ b/libs/ngs-vdb/test/CMakeLists.txt
@@ -70,32 +70,39 @@ add_test ( NAME Test_Ngs COMMAND test-ng
 add_executable ( test-ngs_sra ngstest_sra.cpp )
 target_link_libraries ( test-ngs_sra ${TEST_LIBS})
 add_test ( NAME Test_Ngs_Sra COMMAND test-ngs_sra WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_Sra PROPERTIES LABELS online )
 
 add_executable ( test-ngs_sradb ngstest_sradb.cpp )
 target_link_libraries ( test-ngs_sradb ${TEST_LIBS})
 add_test ( NAME Test_Ngs_SraDb COMMAND test-ngs_sradb WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_SraDb PROPERTIES LABELS online )
 
 add_executable ( test-ngs_csra1 ngstest_csra1.cpp )
 target_link_libraries ( test-ngs_csra1 ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_Csra1 COMMAND test-ngs_csra1 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-set_tests_properties( SlowTest_Ngs_Csra1 PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" )
+set_tests_properties( SlowTest_Ngs_Csra1 PROPERTIES
+  ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS online )
 
 add_executable ( test-ngs_csra1_readcollection ngstest_csra1_readcollection.cpp )
 target_link_libraries ( test-ngs_csra1_readcollection ${TEST_LIBS})
 add_test ( NAME Test_Ngs_Csra1_ReadCollection COMMAND test-ngs_csra1_readcollection WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_Csra1_ReadCollection PROPERTIES LABELS online )
 
 add_executable ( test-ngs_csra1_refwin ngstest_csra1_refwin.cpp )
 target_link_libraries ( test-ngs_csra1_refwin ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_Csra1_RefWin COMMAND test-ngs_csra1_refwin WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-set_tests_properties( SlowTest_Ngs_Csra1_RefWin PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" )
+set_tests_properties( SlowTest_Ngs_Csra1_RefWin PROPERTIES
+                      ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS online )
 
 add_executable ( test-ngs_csra1_pileup ngstest_csra1_pileup.cpp )
 target_link_libraries ( test-ngs_csra1_pileup ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_Csra1_Pileup COMMAND test-ngs_csra1_pileup WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( SlowTest_Ngs_Csra1_Pileup PROPERTIES LABELS online )
 
 add_executable ( test-ngs_reference ngstest_reference.cpp )
 target_link_libraries ( test-ngs_reference ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_Reference COMMAND test-ngs_reference WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( SlowTest_Ngs_Reference PROPERTIES LABELS online )
 
 
 add_executable ( test-fragmentblob ngstest_fragmentblob.cpp )
@@ -103,14 +110,17 @@ target_link_libraries ( test-fragmentblo
 add_test ( NAME SlowTest_Ngs_FragmentBlob COMMAND test-fragmentblob WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 # SlowTest_Ngs_FragmentBlob requires MakeDB to be run first
 set_property(TEST SlowTest_Ngs_FragmentBlob PROPERTY FIXTURES_REQUIRED MakeDB)
+set_tests_properties( SlowTest_Ngs_FragmentBlob PROPERTIES LABELS online )
 
 add_executable ( test-referenceblob ngstest_referenceblob.cpp )
 target_link_libraries ( test-referenceblob ${TEST_LIBS})
 add_test ( NAME Test_Ngs_ReferenceBlob COMMAND test-referenceblob WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_ReferenceBlob PROPERTIES LABELS online )
 
 add_executable ( test-byteblob ngstest_byteblob.cpp )
 target_link_libraries ( test-byteblob ${TEST_LIBS})
 add_test ( NAME Test_Ngs_ByteBlob COMMAND test-byteblob WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_ByteBlob PROPERTIES LABELS online )
 
 #-------------------------------------------------------------------------------
 # tests for the NGS C++ APIs
@@ -118,20 +128,24 @@ add_test ( NAME Test_Ngs_ByteBlob COMMAN
 add_executable ( test-ngs_sra_c++ SRAtest.cpp )
 target_link_libraries ( test-ngs_sra_c++ ${TEST_LIBS})
 add_test ( NAME Test_Ngs_CPP_Sra COMMAND test-ngs_sra_c++ WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_CPP_Sra PROPERTIES LABELS online )
 
 add_executable ( test-ngs_sradb_c++ SRADBtest.cpp )
 target_link_libraries ( test-ngs_sradb_c++ ${TEST_LIBS})
 add_test ( NAME Test_Ngs_CPP_SraDb COMMAND test-ngs_sradb_c++ WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+set_tests_properties( Test_Ngs_CPP_SraDb PROPERTIES LABELS online )
 
 add_executable ( test-ngs_csra1_c++ CSRA1test.cpp )
 target_link_libraries ( test-ngs_csra1_c++ ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_CPP_CSRA1 COMMAND test-ngs_csra1_c++ WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-set_tests_properties( SlowTest_Ngs_CPP_CSRA1 PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" )
+set_tests_properties( SlowTest_Ngs_CPP_CSRA1 PROPERTIES
+                      ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS online )
 
 add_executable ( test-ngs_csra1pileup++ CSRA1PileupTest.cpp )
 target_link_libraries ( test-ngs_csra1pileup++ ${TEST_LIBS})
 add_test ( NAME SlowTest_Ngs_CPP_CSRA1Pileup COMMAND test-ngs_csra1pileup++ WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-set_tests_properties( SlowTest_Ngs_CPP_CSRA1Pileup PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" )
+set_tests_properties( SlowTest_Ngs_CPP_CSRA1Pileup PROPERTIES
+                      ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS online )
 
 #-------------------------------------------------------------------------------
 # tests for this library, ngs-vdb
@@ -146,4 +160,5 @@ target_link_libraries ( test-ngs-vdb
 )
 
 add_test ( NAME SlowTest_NgsVdbWhiteBox COMMAND test-ngs-vdb )
+set_tests_properties( SlowTest_NgsVdbWhiteBox PROPERTIES LABELS online )
 
Index: b/test/external/ngs/testy-the-bear/CMakeLists.txt
===================================================================
--- a/test/external/ngs/testy-the-bear/CMakeLists.txt
+++ b/test/external/ngs/testy-the-bear/CMakeLists.txt
@@ -30,6 +30,6 @@ if( NOT WIN32 )
         COMMAND runtestcase.sh "${TESTBINDIR}/testy-the-bear" "1" "0"
         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
     set_tests_properties( SlowTest_testy-the-bear
-        PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R"
+        PROPERTIES ENVIRONMENT "NCBI_VDB_QUALITY=R" LABELS online
     )
 endif()
Index: b/tools/external/sra-search/test/CMakeLists.txt
===================================================================
--- a/tools/external/sra-search/test/CMakeLists.txt
+++ b/tools/external/sra-search/test/CMakeLists.txt
@@ -73,7 +73,9 @@ add_executable ( test-searchblock
 # white box tests
 add_test ( NAME SearchBlockWhiteBox COMMAND test-searchblock )
 add_test ( NAME SraSearchWhiteBox COMMAND test-sra-search )
+set_tests_properties( SraSearchWhiteBox PROPERTIES LABELS online )
 add_test ( NAME SlowTest_SraSearchWhiteBox COMMAND test-sra-search-slow )
+set_tests_properties( SlowTest_SraSearchWhiteBox PROPERTIES LABELS online )
 
 # command line tests
 if ( NOT WIN32)
